org.eclipse.buildship.ui/src/main/java/org/eclipse/buildship/ui/workspace/AddBuildshipNatureHandler.java

```diff
@@ -58,7 +58,10 @@ public Object execute(ExecutionEvent event) throws ExecutionException {
         for (Object element : elements) {
             IProject project = adapterFunction.apply(element);
             if (project != null && !GradleProjectNature.isPresentOn(project) && project.getLocation() != null) {
-                projects.add(project);
+                IPath location = project.getLocation();
```

Comment:
Something's wrong with this diff, this can't even compile.

Also, the location null check already happens above.